### ISSUE TITLE
Asking for help

### ISSUE THREAD

**Issue by Neil4466** [yassinekebbati/GA-optimized-MLP-based-LPV_MPC#1]
When I tried to launch the script "MAIN_LPV_MPC.m", I got  the following bug:

Dot indexing is not supported for variables of this type.

Error in SeriesNetwork/predict (line 263)
            Y = this.UnderlyingDAGNetwork.predict(X, varargin{:});

Error in pred (line 12)
        pr = net.predict(Zx');

Error in MAIN_LPV_MPC (line 149)
    V = pred([XX(4,1); XX(5,1);UU(2,1);UU(1,1);Wz(1)] , xmean, xstdev, ymean,ystdev, mod);  %use this when using : best_net/test_net

Can you please tell me which version of MATLAB you are using? Actualy, I don't know why this error occured. Can you help me? Thank you!

**Comment by Neil4466** [comment]
My MATLAB version is R2020a.

**Comment by yassinekebbati** [comment]
Hello!If your MATLAB version is at least 2018a or more recent tgen it is okay.Check if you have the deep learning toolbox installed!?I will have a look at the error and let you know.RegardsOn Jan 12, 2024 5:39 AM, Neil4466 ***@***.***> wrote:
My MATLAB version is R2020a.

**Comment by Neil4466** [comment]
Thank you for reply! I have installed the deep learning toolbox before I launched the script.

**Comment by yassinekebbati** [comment]
After looking at the issue, it seems this is due to the Matlab version, it is related to the use of batch normalization in neural network. I do not face this issue on my machine although the version I tested on is 2022b.
you can omit the neural network predictions by commenting lines :145 and 148 to 151 and replacing  lines :146 and 147. (see screenshot below)  to see if the rest of the code works
![MATLAB_Nea37QjXKi](https://github.com/yassinekebbati/GA-optimized-MLP-based-LPV_MPC/assets/49916054/18cd9c25-71f3-4e83-b63e-9959606a4cd3)

**Comment by Neil4466** [comment]
Thank you for your time! I have commented the code you mentioned, and the rest of the code actually works. Your advice helps me a lot.

**Comment by yassinekebbati** [comment]
Happy to help, for the neural network part a more recent version fo Matlab should work fine.
